src/RestApi/StoreApi/Schemas/CartItemSchema.php

```diff
@@ -189,16 +195,24 @@ public function product_id_exists( $product_id ) {
 	public function get_item_response( $cart_item ) {
 		$product = $cart_item['data'];
 
+		$description_char_limit = 75;
+
+		$short_description = $product->get_short_description() ? $product->get_short_description() : wc_trim_string( $product->get_description(), $description_char_limit );
```

Comment:
Probably should pass a context string as an argument to avoid all the default filtering on the short description (maybe `raw` or `edit`?) The default context is [`view`](https://github.com/woocommerce/woocommerce/blob/master/includes/abstracts/abstract-wc-product.php#L246-L248) which in turn leads to [default filtering being applied](https://github.com/woocommerce/woocommerce/blob/master/includes/abstracts/abstract-wc-data.php#L750-L752).

I wonder if this will remove the need for some of the subsequent sanitization happening here (maybe not because we might still be trimming the full description for the product)?

Reply:
Also, I note this implementation varies from that of the `ProductSchema` for the `store` api: 

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/src/RestApi/StoreApi/Schemas/ProductSchema.php#L225

We probably should have parity here.  So instead of `short_description`, this should just be `description`.  Also maybe just follow the same logic as the `ProductSchema` (except I think my comments about the context argument probably should apply to both implementations)?

Reply:
Thanks for the tip about context and filtering – I'll take a look at that and I'll do the same on products endpoint (in this PR too). I agree that these two APIs should use same approach for retrieving/rendering the data.

> We probably should have parity here. So instead of `short_description`, this should just be `description`.

This difference is intentional and necessary in my view. This cart line item `short_description` is displayed in a relatively small area of the cart; I'm guessing that the product description doesn't have that limitation. 

We're intentionally limiting this to 75 chars for the cart design, and I don't believe we should limit the products endpoint in the same way (though .. it depends on what we intend to use it for). See @garymurray comment above: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1552#issuecomment-573719469

Reply:
If we do render the full description for cart/items, potentially we could truncate it on the client. I haven't really considered that approach. Generally I default to the server handling this kind of data-massaging.

Reply:
Moving this out to a separate comment below ..

Reply:
> potentially we could truncate it on the client.

I think any truncation should be client handled. Client should decide how to display/use data, not server (when it comes to data provided through a REST api).

> I tried using edit and view explicitly when calling get_short_description()...It looks like it's a REST API param which indicates where the data will be used; e.g. for viewing, or for editing.

To be clear, I'm wasn't specifically talking about the same thing as `context` in the REST endpoints although that's likely why it's present as a `short_description` method argument.  I'm specifically talking about it as an argument on `WC_Product::short_description`.  Following the code chain, leads to `WC_Product::get_prop` which as I pointed out in the [code I linked to](https://github.com/woocommerce/woocommerce/blob/master/includes/abstracts/abstract-wc-data.php#L750-L752) means that when the `view` context value is used (which happens to be the default for the argument in the case of `short_description`), filters are applied on the description.

Your tests seem to indicate there are no _default_ filters from WC core hooked into this specific filter (which is great!), so we just need to decide whether we care about whether we should expose _filtered_ content, or non filtered content and if we pass through what context might have been specified in the request or not.

> There's also the possibility that edit would require edit permissions, though this doesn't seem to be an issue (tested with incognito window, short description still visible with $context=edit) (perhaps woo doesn't enforce that?).

Whether there are capability checks applied to specific context views (in the case of the REST API) depends on if the endpoint implements that.  Currently the store endpoints have no capability restrictions because of the nature of the endpoint which is for public consumption.

> Should product description use the main product content field (i.e. the_post)?
> Should product description render shortcodes and include markup, or should it be plain text?

Regarding these two points,   `post_content` exposes `raw` and `rendered` values for responses on the `posts/` REST endpoint.  However, raw is only exposed on authed edit context responses because it is the non-rendered actual content (typically used in the editor). In our case, being this is a _public_ endpoint not intended for editor use, but only for viewing data, I think we should consider whether we just return the full rendered value and leave it up to the client whether it decides to strip any html etc before rendering.  The _rendered_ value _should_ be safe if it goes through the same WP sanitization that happens when prepping content for output.

Alternatively we could use `rendered` and `stripped` properties here, so that the server provides both fully rendered and fully stripped values for the `description` (or `short_description`) which allows the client to choose which to work from.

Reply:
Just to add, if not already mentioned, `description` from the Product StoreAPI is the "short description" from woo core, or a truncated (400 chars) long description if not yet.

In the atomic blocks, "summary" block, it takes this and further truncates it with a 'read more' button on the client side. I think thats what @nerrad is suggesting? Take this field and apply more truncation if needed. 

See the`ProductSummary` component.

I don't think it would be appropriate to return the entire post content in this API, but I see how description and short description could be confused. With that in mind, perhaps we should rename the field `summary`, since in my mind it sounds more like what is returned than `short description`. We don't need to follow core naming convention in this API. 

As for context, I don't think it's handled in any API. The returned content is always the same.

Reply:
OK so I've now removed the custom truncation: e8d0d781e6cd3556b2c6f67cd44b5299b61159a0

I've left the html + shortcode stripping, as this value (cart-items short description) needs to be plain text. The products API value is [displayed in different context](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/d535b4d9c4daf14f00813aeb05268e7f805c719a/assets/js/atomic/components/product/summary/index.js#L20-L22) and is expected to contain markup.

We'll need to remember to implement the truncation on the client when we hook up the cart to the API, so I've added a note to #1489

@nerrad I'm thinking this is good to merge now - anything else?